src/ui/simulator/toolbox/components/datagrid/renderer/area/renewable.areasummary.cpp

```diff
@@ -0,0 +1,256 @@
+    //  to fix the min width of the column. A method should be added for this purpose.
+    static const wxChar* const rows[] = {
+      wxT("          Group          "),
+      wxT("  Enabled  "),
+      wxT("  Unit count  "),
+      wxT("   Nom. Capacity  \n        MW        "),
+    };
+    return (colIndx < 5 and colIndx >= 0) ? rows[colIndx] : wxT("");
+}
+
+wxString RenewableClusterSummarySingleArea::cellValue(int x, int y) const
+{
+    Data::RenewableCluster* cluster = (pArea and (uint) y < pArea->renewable.list.size())
+                                      ? pArea->renewable.list.byIndex[y]
+                                      : nullptr;
+    // gp : do we wish to have the line empty if cluster disabled
+    // if (!cluster->enabled)
+    //    return wxEmptyString;
+    switch (x)
+    {
+    case 0:
+        return wxStringFromUTF8(cluster->group());
+    case 1:
+        return cluster->enabled ? wxT("Yes") : wxT("no");
+    case 2:
+        return wxString() << cluster->unitCount;
+    case 3:
+        return DoubleToWxString(cluster->nominalCapacity);
+    }
+    return wxEmptyString;
+}
+
+double RenewableClusterSummarySingleArea::cellNumericValue(int x, int y) const
+{
+    Data::RenewableCluster* cluster = (pArea and (uint) y < pArea->renewable.list.size())
+                                      ? pArea->renewable.list.byIndex[y]
+                                      : nullptr;
+    // gp : do we wish to have the line empty if cluster disabled
+    // if (!cluster->enabled)
+    //     return 0.;
+    switch (x)
+    {
+    case 0:
+        return 0.;
+    case 1:
+        return cluster->enabled ? 1. : 0.;
+    case 2:
+        return cluster->unitCount;
+    case 3:
+        return cluster->nominalCapacity;
+    }
+    return 0.;
+}
+
+void RenewableClusterSummarySingleArea::onAreaChanged(Antares::Data::Area* area)
+{
+    if (pArea != area)
+    {
+        pArea = area;
+        RefreshAllControls(pControl);
+    }
+}
+
+IRenderer::CellStyle RenewableClusterSummarySingleArea::cellStyle(int col, int row) const
+{
+    return (col > 0 and Math::Zero(cellNumericValue(col, row)))
+             ? IRenderer::cellStyleDefaultDisabled
+             : (col == 1 || col == 2) ? IRenderer::cellStyleConstraintWeight
+                                      : IRenderer::cellStyleDefault;
+}
+
+struct NoCheck
+{
+    template<class T>
+    static bool Validate(const T&)
+    {
+        return true;
+    }
+};
+
+struct CheckUnitCount
+{
+    static bool Validate(uint& f)
+    {
+        if (f > 100)
+            f = 100;
+        return true;
+    }
+};
+
+template<class CheckT>
+static bool UpdateUnsignedLong(uint& value, const String& str)
+{
+    uint l;
+    if (str.to(l))
+    {
+        if (value != l and CheckT::Validate(l))
+        {
+            value = l;
+            MarkTheStudyAsModified();
+            OnInspectorRefresh(nullptr);
+            return true;
+        }
+    }
+    return false;
+}
```

Comment:
It would be nice if class CheckT had this function, since it is generic *ie* not specific to renewable area summary.

It would need a callback to call when a valid expression is provided.

```cpp
template<class T, class predicateT>
checkAndValidate(T value, predicateT callback)
{
    T l;
    if (str.to<T>(l))
    {
        if (value != l and CheckT::Validate(l))
        {
            value = l;
            callback();
            return true;
        }
    }
    return false;
}

void cb() {
            MarkTheStudyAsModified();
            OnInspectorRefresh(nullptr);
}

checkAndValidate<uint, void(*)(void)>(value, cb);
```